include/flatcc/portable/pdiagnostic_pop.h

```diff
@@ -1,15 +1,25 @@
 #ifndef PDIAGNOSTIC_POP_H
 #define PDIAGNOSTIC_POP_H
 
+#if defined(PDIAGNOSTIC_PUSHED_MSVC)
 #if PDIAGNOSTIC_PUSHED_MSVC
 #pragma warning( pop )
+#endif // PDIAGNOSTIC_PUSHED_MSVC
 #undef PDIAGNOSTIC_PUSHED_MSVC
```

Comment:
We should probably define it to 0 after undef so the multiple pops can rely on the flag being present. Likewise for subsequent code.

Reply:
The if-defined checks you commented about were to take care of this, since the pre-existing code was using undef and I wanted to disrupt as little as possible.

I have no preference here either, but one thing that is potentially a little better about using undef is that after all the flatcc headers are done, these names will no longer exist in the preprocessor environment, which feels slightly cleaner to me.

Reply:
yes, regarding the pop cleanup, that is is a valid point - but there are still other flags anyway.

Reply:
It's not clear to me; do you want this changed?  I have no preference, but I've already spent far more time on all of this than I had intended, and I may soon just decide to hack around these issues in my own build.

Reply:
I't too tired to have an opnion right now though it would be nice if the pop logic could be more elegant / less bulky. Mostly I want consistency.

Reply:
No problem, I'm pretty tired myself.  If you decide there's a specific change you want me to make, I'm happy to make it, but I can't really justify (to my employer) taking too much additional time to restyle code.